### PR TITLE
chore: improve tracer time by consuming as events

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ Changes the Tracing.start protocol command to use the `ReportEvents` instead of `ReturnAsStream`. This makes the trace gathering much faster, as observed in some automated run timings:

### Average timings - With filmstrips alone
Before - 3 seconds 
After -  2.5 seconds 

This would be useful as we add more categories to trace when we start doing this #120

### CODE REVIEW

File: src/plugins/tracing.ts [hunk 1]
```diff
@@ -80,26 +61,34 @@ export function filterFilmstrips(buffer: Buffer): Array<FilmStrip> {
...
-
+    const includedCategories = [
+      // exclude all default categories
+      '-*',
+      // capture screenshots
+      'disabled-by-default-devtools.screenshot',
+    ];
     await client.send('Tracing.start', {
-      transferMode: 'ReturnAsStream',
+      transferMode: 'ReportEvents',
```
Comment (jahtalab) [inline src/plugins/tracing.ts:71]: The transferMode `ReportEvents` is the default mode, so we can remove this.
also, might be worth adding a comment in the code to explain choosing the default mode over `ReturnAsStream`

**Review by jahtalab — COMMENTED** [review]
Thanks @vigneshshanmugam , generally looks good, added a few comments.

File: src/plugins/tracing.ts (hunk 1 above)
Comment (vigneshshanmugam) [reply src/plugins/tracing.ts:71]: Yeah its the default, i wanted to keep it as they are deprecating some of the fields and might change the default anytime. I will add a comment though about the perf implication.

Comment (jahtalab) [reply src/plugins/tracing.ts:74]: Out of interest, do you have a link for the that announcement?

**Review by jahtalab — APPROVED** [review]

File: src/plugins/tracing.ts (hunk 1 above)
Comment (vigneshshanmugam) [reply src/plugins/tracing.ts:74]: No i meant the deprecation notices in the latest tip of tree in CDP https://chromedevtools.github.io/devtools-protocol/tot/Tracing/#method-start. None of them are at the stable RC at the moment. Anything could change, so this was just us being defensive, nothing announced yet.

**Pull request merged**